### ISSUE TITLE
sys.settrace doesn't disable tracing if a local trace function returns None

### ISSUE THREAD

**Issue by nedbat** [python/cpython#56201]
BPO | [11992](https://bugs.python.org/issue11992)
--- | :---
Nosy | @arigo, @nedbat, @xdegaye

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/nedbat'
closed_at = None
created_at = <Date 2011-05-04.01:58:53.007>
labels = ['type-bug', 'library']
title = "sys.settrace doesn't disable tracing if a local trace function returns None"
updated_at = <Date 2017-01-18.11:32:14.531>
user = 'https://github.com/nedbat'
```

bugs.python.org fields:
```python
activity = <Date 2017-01-18.11:32:14.531>
actor = 'arigo'
assignee = 'nedbat'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2011-05-04.01:58:53.007>
creator = 'nedbat'
dependencies = []
files = []
hgrepos = []
issue_num = 11992
keywords = []
message_count = 3.0
messages = ['135089', '206739', '285720']
nosy_count = 3.0
nosy_names = ['arigo', 'nedbat', 'xdegaye']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue11992'
versions = ['Python 2.7', 'Python 3.2']
```

</p></details>

**Comment by amakelov** [comment]
Hi, I've been trying to turn off tracing selectively in certain functions (e.g., stop tracing calls when they invoke functions defined outside of a certain directory), and running into this problem in Python 3.10 as well. Are you aware of any workaround that would enable this sort of logic?

**Comment by brandtbucher** [comment]
Fixed the docs in [`d9246c7`](https://github.com/python/cpython/commit/d9246c7b734b8958da03494045208681d95f5b74).
